hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ReadOnlyZKClient.java

```diff
@@ -274,6 +309,28 @@ protected void doExec(ZooKeeper zk) {
     return future;
   }
 
+  public CompletableFuture<Stat> existsWithTimeout(String path, int endTime) {
+    CompletableFuture<Stat> future = exists(path);
+    TimerTask timerTask = new TimerTask() {
+      @Override
+      public void run(Timeout timeout) throws Exception {
+        if (EnvironmentEdgeManager.currentTime() > endTime) {
```

Comment:
Better use System.nanoTime instead of currentTime. And for Betty's timer, I do not think we need to check for endTime again, we should queue a task into the queue to finish the future without timeout exception.